gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationViewModel.java

```diff
@@ -316,7 +320,15 @@ public void onSaveClick() {
 
   private void onObservationLoaded(Observation observation) {
     this.originalObservation = observation;
-    refreshResponseMap(observation);
+
+    // Photo field is updated by launching an external intent. This causes the form to reload.
+    // When that happens, we don't want to lose the unsaved changes.
```

Comment:
Wouldn't this be equivalent to opening another app while editing the form? If so, how is this working today for non-photo fields, and why are photos different?

Reply:
The reason is we handle the result of the photos before the response map is refreshed. As mentioned in the previous comment, this is a side-effect of an earlier pending refactor.

Reply:
If the `initialize` is getting called whenever we reopen the Fragment, I'm not sure how this is actually working today.  There are many reasons the fragment could get reinitialized; it's hard to name `isPhotoFieldUpdated` appropriately because it's not really well defined - iiuc it's asking "am I being reinitialized because the user left to a photo intent and came back?" In reality I don't think we can know that for sure, and as result there are likely many edge cases where this will fail.

I may be a bit out of my depth on Android fragment/VM lifecycles, but I'll approve for now to unblock current work. @dturner, would you mind taking a look at the `EditObservation*` logic when you get a chance to see how we might improve?